src/apps/service/apps/jobs/sharing.clj

```diff
@@ -150,42 +161,94 @@
   (process-job-inputs (partial share-input-file sharer sharee) apps-client job))
 
 (defn- share-job
-  [apps-client sharer sharee {job-id :analysis_id level :permission}]
-  (if-let [job (jp/get-job-by-id job-id)]
+  [update-fn apps-client sharer sharee {job-id :analysis_id level :permission}]
+  (let [job-id (uuidify job-id)
+        job    (jp/get-job-by-id job-id)]
     (try+
-     (verify-not-subjob job)
-     (verify-accessible sharer job-id)
-     (verify-support apps-client job-id)
-     (verify-not-group sharee job-id)
-
      (share-analysis job-id sharee level)
 
      (let [child-input-share-errs (process-child-jobs (partial share-child-job apps-client sharer sharee level) job-id)
            input-share-errs       (process-job-inputs (partial share-input-file sharer sharee) apps-client job)
            output-share-err-msg   (share-output-folder sharer sharee job)
            app-share-err-msg      (share-app-for-job apps-client sharer sharee job-id job)]
+       (update-fn (format "shared job ID %s with %s" job-id sharee))
        (job-sharing-success job-id
                             job
                             level
                             (concat input-share-errs child-input-share-errs)
                             output-share-err-msg
                             app-share-err-msg))
      (catch [:type ::job-sharing-failure] {:keys [failure-reason]}
+       (update-fn (format "failed to share job ID %s with %s: %s" job-id sharee failure-reason))
        (job-sharing-failure job-id job level failure-reason))
-     (catch [:type ::permission-load-failure] {:keys [reason]}
```

Comment:
`::permission-load-failure` wasn't being thrown anywhere, so i removed this catch block.